### DISCUSSION TITLE
Aligning modules to the right side of the shell.

### DISCUSSION THREAD

**Discussion by Malternative3772** [starship/starship#2046] [Q&A] [answered]
Hello,
I just wanted to ask if there is a way to align modules to the right side of the shell, so that it looks something like this:
```
`user@device in ~/                                                                                 9:35:29
 > `
```
This would allow me to have things like the battery usage and the date aligned on the right of the terminal window, and to have important information the username, hostname and directory on the left side.

**Comment by chipbuster** [comment]
This is a popular feature request (tracking issue for it is https://github.com/starship/starship/issues/498) but we haven't had anyone who's been willing to implement it so far.

**Comment by mickimnet** [comment]
@Malternative3772 what shell are you using? There is an easy way to do it in `Fish Shell`, with a second `starship.toml` File for the right side.

**Reply by Malternative3772** [reply]
Yes, I am using the Fish shell. Thank you very much. But how do I set up Starship to display the second prompt on the right side. Do I just add a second  `starship init fish | source` into the `fish.conf` file?

**Comment by mickimnet** [comment, marked as answer]
As I believe the questions will be asked by quite some people, and I wanted to get the _right prompt_ to work in `Zsh` anyway, here is the guide to set up a _right prompt_ in **both** shells.

# Workaround for Right Prompt (Zsh & Fish Shell)

- **First**, let me start with a big **disclaimer**: anything you're doing here `may break with any update of` _Starship_, as we may change the init process of _Starship_. Therefore you need to check it on **each update** of _Starship_.

- **Second**, as prerequisite, you should feel comfortable fiddling around with your shell configuration as well as know at least a bit about scripting in your shell (`Fish Shell` or `Zsh`).

## General notes

You need to run `starship init $YourShell --print-full-init` and catch that output. If an update of _Starship_ comes out, you need to compare the output of the new version to your saved output and update it accordingly.

Also, you need to create two `starship.toml` files, one for the left and one for the right configuration (see our [documentation](https://starship.rs/config/) for how to do that). I prefer to name them `starship-left.toml` and `starship-right.toml`, and will use those names in this guide.

## Fish shell

Run `starship init fish --print-full-init` and catch that output. You basically have two sections in the output. The first is starting with:

```fish
function fish_prompt
  ...
end
```

The second section starts after the `end` and needs to go in your `config.fish` (and replace the occurrence of `starship init fish | source `).

The first sections needs to go in your `fish_prompt` function (you can run `funced fish_prompt` to edit it) and you need to add the following line **above** the `"/Your/Local/bin/Path/to/starship" prompt ...` line.
```fish
  ...
  set --export STARSHIP_CONFIG /Your/Path/To/Your/starship-left.toml    ## change /Your/Path/To/Your/starship-left.toml
  "/Your/Local/bin/Path/to/starship" prompt --status=$exit_code --keymap=$keymap --cmd-duration=$starship_duration --jobs=(count (jobs -p))    ## Don't copy this line! It's just for reference ;)
  ...
```

And finally, you need to edit the `fish_right_prompt` function (you can also use `funced fish_right_prompt` to edit it).
```fish
function fish_right_prompt --description 'Write out the prompt on the right side (starship style)'
  set --export STARSHIP_CONFIG Your/Path/To/Your/starship-right.toml    ## change Your/Path/To/Your/starship-right.toml
  "/Your/Local/bin/Path/to/starship" prompt
end
```
If you want to have the `status`, the `duration` or the `jobs` module to be displayed on the right side, you need to change the `"/Your/Local/bin/Path/to/starship" prompt`-line accordingly.

## Zsh

Run `starship init zsh --print-full-init >/Your/Path/.zsh_prompt`. That stores the init code into a new file, which we're going to edit.

When you open that file in your editor, you need to replace the following line starting with `PROMPT=...`  and add the `RPROMPT=...` line below it:
```zsh
...
starship_render() {
  ...
  ## change /Your/Path/To/Your/starship-left.toml
  PROMPT="$(STARSHIP_CONFIG=/Your/Path/To/Your/starship-left.toml "/Your/Local/bin/Path/to/starship" prompt --keymap="${KEYMAP-}" --status=$STARSHIP_CMD_STATUS --cmd-duration=${STARSHIP_DURATION-} --jobs="$NUM_JOBS")"
  ## change /Your/Path/To/Your/starship-right.toml
  RPROMPT="$(STARSHIP_CONFIG=/Your/Path/To/Your/starship-right.toml "/Your/Local/bin/Path/to/starship" prompt)"
  ...
}
...
```
If you want to have the `status`, the `duration` or the `jobs` module to be displayed on the right side, you need to change the `"/Your/Local/bin/Path/to/starship" prompt`-line accordingly.

Finally you need to change the line `eval "$(starship init zsh)"` in your Zsh configuration file (e. g. `~/.zshrc`) to `source /Your/Path/.zsh_prompt`.

Take care
Mick
